src/lib/to-facade.ts

```diff
@@ -14,7 +14,55 @@ export type SegmentFacade = Facade<SegmentEvent> & {
   obj: SegmentEvent
 }
 
+interface Properties {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [key: string]: any
+}
+
+interface ClonedObj {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [key: string]: any
+}
+
+const convertClassesToObjects = (
+  properties: Properties,
+  tmp: ClonedObj = {}
+) => {
+  if (typeof properties !== 'object') return properties
```

Comment:
It looks like this function doesn't properly handle arrays. 
I ran a test where one of the properties was an array:
```js
await analytics.track(
        'event',
        { prop: true, prop2: stubbedClass, list: ['a', 2, {foo: 'bar'}] },
        { opt: true }
      )
```
Before this change, the array was copied over correctly, but after this change I see `list: {}` instead. I think an `Array.isArray` check where you call `convertClassesToObjects` on each element _should_ work.

Reply:
noticing the same with the qa tests. I'll look into it